scripts/apps/monitoring/directives/MonitoringView.js

```diff
@@ -14,20 +14,65 @@ export function MonitoringView($rootScope, authoringWorkspace) {
             state: '='
         },
         link: function(scope, elem) {
+
             var containerElem = elem.find('.content-list');
             containerElem.on('scroll', handleContainerScroll);
 
-            function handleContainerScroll() {
+            scope.viewColumn = scope.monitoring.viewColumn;
+
+            scope.displayColumn = function() {
+                scope.viewColumn = !scope.viewColumn;
+                scope.monitoring.switchView(scope.viewColumn);
+                $rootScope.$broadcast('view:column', {viewColumn: scope.viewColumn});
+            };
+
+            scope.isActiveGroup = function(group) {
+                return scope.monitoring.selectedGroup ? (scope.monitoring.selectedGroup._id === group._id) : true;
+            };
+
+            var updateTimeout;
+            function handleContainerScroll($event) {
                 if ($rootScope.itemToogle) {
                     scope.$applyAsync(function() {
                         $rootScope.itemToogle(false);
                         $rootScope.itemToogle = null;
                     });
                 }
+
+                // If scroll bar leaves top position update scope.scrollTop
+                // which is used to display refresh button on list item updates
+                if ($event.currentTarget.scrollTop >= 0 && $event.currentTarget.scrollTop < 100) {
+                    scope.$applyAsync(function() {
+                        scope.scrollTop = scope.monitoring.scrollTop = $event.currentTarget.scrollTop;
+
+                        // force refresh the group or list, if scroll bar hits the top of list.
+                        if (scope.monitoring.scrollTop === 0) {
+                            $rootScope.$broadcast('refresh:list', scope.group);
+                        }
+                    });
+                }
+
+                $timeout.cancel(updateTimeout);
+                updateTimeout = $timeout(renderIfNeeded($event), 100, false);
+            }
+
+            function isListEnd(containerElem) {
+                return containerElem.scrollTop + containerElem.offsetHeight + 200 >= containerElem.scrollHeight;
+            }
+
+            /**
+             * Trigger render in case user scrolls to the very end of list
+             */
+            function renderIfNeeded($event) {
+                if (isListEnd($event.currentTarget)) {
+                    scope.rendering = scope.loading = true;
+                    $rootScope.$broadcast('render:next');
```

Comment:
imo this could be `scope.$broadcast` to make it more obvious that it targets some component inside